pkg/operator2/configobservation/oauth/observe_tokenconfig.go

```diff
@@ -0,0 +1,89 @@
+package oauth
+
+import (
+	"k8s.io/klog"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation"
+)
+
+const defaultAccessTokenMaxAgeSeconds = float64(86400) // a day
+
+func ObserveTokenConfig(genericlisters configobserver.Listers, recorder events.Recorder, previouslyObservedConfig map[string]interface{}) (map[string]interface{}, []error) {
+	tokenConfigPath := []string{"oauthConfig", "tokenConfig"}
+	listers := genericlisters.(configobservation.Listers)
+	errs := []error{}
+
+	currentTokenConfig, _, err := unstructured.NestedMap(previouslyObservedConfig, tokenConfigPath...)
+	if err != nil {
+		return previouslyObservedConfig, append(errs, err)
+	}
+
+	currentAccessTokenMaxAgeSecondsFloat, _, err := unstructured.NestedFloat64(currentTokenConfig, "accessTokenMaxAgeSeconds")
+	if err != nil {
+		errs = append(errs, err)
+	}
+	currentAccessTokenMaxAgeSeconds := float64(currentAccessTokenMaxAgeSecondsFloat)
+
+	currentAccessTokenInactivityTimeoutSeconds, _, err := unstructured.NestedFloat64(currentTokenConfig, "accessTokenInactivityTimeoutSeconds")
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	observedConfig := map[string]interface{}{}
+	oauthConfig, err := listers.OAuthLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		klog.Warning("oauth.config.openshift.io/cluster: not found")
+		return map[string]interface{}{
+			"oauthConfig": map[string]interface{}{
+				"tokenConfig": map[string]interface{}{
+					"accessTokenMaxAgeSeconds": defaultAccessTokenMaxAgeSeconds,
+				},
+			},
+		}, nil
+	} else if err != nil {
+		return previouslyObservedConfig, append(errs, err)
+	}
+
+	observedAccessTokenMaxAgeSeconds := float64(oauthConfig.Spec.TokenConfig.AccessTokenMaxAgeSeconds)
+	if observedAccessTokenMaxAgeSeconds == 0 {
+		observedAccessTokenMaxAgeSeconds = defaultAccessTokenMaxAgeSeconds
+	}
+
+	observedTokenConfigFieldMap := map[string]interface{}{}
+	unstructured.SetNestedField(observedTokenConfigFieldMap, observedAccessTokenMaxAgeSeconds, "accessTokenMaxAgeSeconds")
+
+	observedAccessTokenInactivityTimeoutSeconds := float64(oauthConfig.Spec.TokenConfig.AccessTokenInactivityTimeoutSeconds)
```

Comment:
It appears that this switch statement is just ensuring a value less than 0 is reset to zero, so maybe simplify as follows?

```golang
if val < 0 {
  val = 0 
}
fieldMap["val"] = val
```